dynparquet/record_builder.go

```diff
@@ -0,0 +1,597 @@
+package dynparquet
+
+import (
+	"reflect"
+	"slices"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/apache/arrow/go/v14/arrow"
+	"github.com/apache/arrow/go/v14/arrow/array"
+	"github.com/apache/arrow/go/v14/arrow/memory"
+	"github.com/google/uuid"
+	schemapb "github.com/polarsignals/frostdb/gen/proto/go/frostdb/schema/v1alpha1"
+)
+
+const (
+	TagName = "frostdb"
+)
+
+// Build is a generic arrow.Record builder that ingests structs of type T. The
+// generated record can be passed to (*Table).InsertRecord.
+//
+// Struct tag `frostdb` is used to pass options for the schema for T and use
+// (*Build[T]).Schema to obtain schema v1alpha1.
+//
+// This api is opinionated.
+//
+//   - Dynamic columns are supported but only for  maps and struct slices
+//
+//   - Repeated Columns are not supported
```

Comment:
Is this a fundamental issue or just not something that is supported today?

Reply:
So I asked because `v1alpha1` technically does support the repeated column type using https://github.com/polarsignals/frostdb/blob/bc23e42eda40e0d86d629aa7837e23bfa52f5b2c/proto/frostdb/schema/v1alpha1/schema.proto#L90

Reply:
Oops! I missed that. Still, we probably need to keep this PR simple  for reviewing. I will prepare a follow up patch when this lands.

Unless if this is necessary. My focus was to handle https://github.com/parca-dev/parca/blob/main/pkg/profile/schema.go which doesn't have repeated fields.

Reply:
@thorfour since you will be the one reviewing, do you think I should add repeated fields here or on a followup PR?

If here, I was wondering how repeated field behaves for dynamic columns ?

Reply:
We can do it in a follow up that's no problem. Just wanted to raise it here so that it was documented.

I think we'd want to slightly change how dynamic columns are handled. I think we should probably require them to be of type `map[string]any` where the key in the map would be the fully qualified dynamic column name. 

So we'd want to change `Labels` in the `Sample` struct into 
```go
type Sample struct {
...
    Labels map[string]string
...
}
```

Which I think is probably better behavior anyways, it was kind of a strange thing to assume that the `Name` field of the struct should exist for a dynamic column.